src/app.ts

```diff
@@ -25,6 +27,23 @@ import { FHIRStructureDefinitionRegistry } from './registry';
 
 const configVersionSupported: ConfigVersion = 1;
 
+function prepareRequestContext(req: express.Request): RequestContext {
+    const requestContext: RequestContext = {
+        headers: req.headers,
+        hostname: req.hostname,
+        url: req.url,
+        contextInfo: {},
+    };
+
+    if (req.method) {
+        const method = req.method.toUpperCase();
+        if (['CONNECT', 'DELETE', 'GET', 'HEAD', 'OPTIONS', 'PATCH', 'POST', 'PUT', 'TRACE'].includes(method)) {
+            requestContext.verb = method as VerbType;
+        }
+    }
```

Comment:
Anyway this `if` doesn't pass? can `req.method` be `null`? Same question for the types of methods?

Reply:
Yes, technically the interface allows the `verb` to be undefined. That's because `req.method`'s type is a string but for `verb` only the union type values are allowed. In real life `req.method` should always have a valid value. In any case the auth code could deny a request without `verb` in the RequestContext.